src/rctmon/mqtt.py

```diff
@@ -10,20 +10,23 @@
 
 class MqttClient():
 
-    mqtt_host = None
-    mqtt_port = None
-
-    mqtt_client = None
-
     def __init__(self, mqtt_config: MqttConfig):
-        self.mqtt_host = mqtt_config.mqtt_host
-        self.mqtt_port = mqtt_config.mqtt_port
-
-        self._connect()
-
-    def _connect(self):
-        self.mqtt_client = mqtt.Client()
-        self.mqtt_client.connect(self.mqtt_host, self.mqtt_port)
+        self.conf = mqtt_config
+        self.mqtt_client = self._connect()
+
+    def _connect(self) -> 'MqttClient':
+        mqtt_client = mqtt.Client()
+        if self.conf.user and self.conf.auth_pass:
```

Comment:
```suggestion
        if self.conf.auth_user and self.conf.auth_pass:
```

This would be auth_user according to the config.

Reply:
is fixed now. Should have done better testing. Which I hopfuly did now.

Reply:
👍